spec/mongoid/association/referenced/has_many/enumerable_spec.rb

```diff
@@ -201,6 +201,122 @@
     end
   end
 
+  describe "#empty?" do
+
+    let(:person) do
+      Person.create
+    end
+
+    let!(:post_one) do
+      Post.create(person_id: person.id)
+    end
+
+    let!(:post_two) do
+      Post.create(person_id: person.id)
+    end
+
+    context "when only a criteria target exists" do
+
+      let(:criteria) do
+        Post.where(person_id: person.id)
+      end
+
+      let!(:enumerable) do
+        described_class.new(criteria)
+      end
+
+      let!(:empty) do
+        enumerable.empty?
+      end
+
+      it "returns false" do
+        expect(empty).to be false
+      end
+
+      it "retains the correct length" do
+        expect(enumerable.length).to eq(2)
+      end
+
+      it "retains the correct length when calling to_a" do
+        expect(enumerable.to_a.length).to eq(2)
+      end
+
+      context "when iterating over the relation a second time" do
```

Comment:
We use "association" because that is the correct term. "Relation" (per the original relational DB model) refers to values in the columns being related into a single table.  For BC reasons the method names retain "relation" in a number of places, but all comments, local variables, prose etc. should use "association".

Reply:
Again, I've just copy-pasted from existing specs. I will do the change here, but I recommend you raise a PR which does this change uniformly across the entire repo.

Reply:
Changed in this PR.